src/App.tsx

```diff
@@ -8,53 +8,125 @@
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { useEffect, useState } from 'react';
+import { Post } from './types/Post';
+import { getPosts } from './api/api';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [openedPost, setOpenedPost] = useState<Post | null>(null);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const selectUser = (user: User) => {
+    setOpenedPost(null);
+    setPosts([]);
+    setSelectedUser(user);
+  };
 
-              <Loader />
+  const loadPost = () => {
+    if (selectedUser && !isError) {
+      setIsLoading(true);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+      getPosts(selectedUser.id)
+        .then(result => {
+          const results = result.map(post => ({
+            id: post.id,
+            userId: post.userId,
+            title: post.title,
+            body: post.body,
+          }));
+
+          setPosts(results);
+        })
+        .catch(() => {
+          setIsError(true);
+          setTimeout(() => setIsError(false), 3000);
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    }
+  };
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+  useEffect(() => {
+    loadPost();
+  }, [selectedUser, isError]);
```

Comment:
Including `isError` in the dependency array of `useEffect` might cause unnecessary re-fetching of posts when the error state changes. Consider removing `isError` from the dependencies to prevent this.